service-commands/scripts/allClockValues.js

```diff
@@ -0,0 +1,230 @@
+  for (let offset = 0; moreTracksRemaining; offset += batchSize) {
+    console.time(`Fetching tracks (${offset} - ${offset + batchSize})`)
+
+    const tracksBatch = (
+      await makeRequest({
+        method: "get",
+        url: "/tracks",
+        baseURL: discoveryProvider,
+        params: { offset, limit: batchSize },
+      })
+    ).data.data
+    moreTracksRemaining = tracksBatch.length === batchSize
+
+    tracksBatch.forEach(({ metadata_multihash, owner_id }) => {
+      trackCids[owner_id] = trackCids[owner_id] || []
+      trackCids[owner_id].push(metadata_multihash)
+    })
+
+    console.timeEnd(`Fetching tracks (${offset} - ${offset + batchSize})`)
+  }
+
+  return trackCids
+}
+
+/**
+ * @param {string} creatorNode - Creator Node endpoint
+ * @param {Array<string>} walletPublicKeys
+ * @returns {Array<Object>} clockValues
+ */
+async function getClockValues(creatorNode, walletPublicKeys) {
+  try {
+    return (
+      await makeRequest({
+        method: "post",
+        url: "/users/batch_clock_status",
+        baseURL: creatorNode,
+        data: { walletPublicKeys },
+      })
+    ).data.users
+  } catch (e) {
+    console.log(`Got ${e} when fetching clock values from ${creatorNode}`)
+    return walletPublicKeys.map((walletPublicKey) => ({
+      walletPublicKey,
+      clock: 0,
+    }))
+  }
+}
+
+/**
+ * @param {string} creatorNode - Creator Node endpoint
+ * @param {Array<string>} cids
+ * @returns {Array<Object>} cidsExist
+ */
+async function getCidsExist(creatorNode, cids) {
+  try {
+    return (
+      await makeRequest({
+        method: "post",
+        url: "/batch_cids_exist",
+        baseURL: creatorNode,
+        data: { cids },
+      })
+    ).data.cids
+  } catch (e) {
+    console.log(`Got ${e} when checking if cids exist in ${creatorNode}`)
+    return cids.map((cid) => ({ cid, exists: false }))
+  }
+}
+
+async function run() {
+  const discoveryProvider = "https://discoveryprovider.staging.audius.co/"
+  // const discoveryProvider = "http://localhost:5000"
+  const trackBatchSize = 500
+  const userBatchSize = 500
+
+  const trackCids = await getTrackCids(discoveryProvider, trackBatchSize)
+
+  let moreUsersRemaining = true
+  for (let offset = 0; moreUsersRemaining; offset += userBatchSize) {
+    console.time(`${offset} - ${offset + userBatchSize}`)
+
+    const usersBatch = await getUsersBatch(
+      discoveryProvider,
+      offset,
+      userBatchSize
+    )
+    moreUsersRemaining = usersBatch.length === userBatchSize
+
+    const creatorNodes = new Set()
+    const cn2wallets = {} // creator node -> wallets
+    const cn2cids = {} // creator node -> cids
+    const cids = {} // user id -> cids
+    usersBatch.forEach(
+      ({
+        creator_node_endpoint,
+        user_id,
+        wallet,
+        cover_photo_sizes,
+        profile_picture_sizes,
+        metadata_multihash,
+      }) => {
+        cids[user_id] = Array.from(trackCids[user_id] || [])
```

Comment:
why not just query at this point in time for trackCIDs associated w/this user?

Reply:
https://github.com/AudiusProject/audius-protocol/blob/master/discovery-provider/src/queries/get_tracks.py#L32

Reply:
Querying the `get_tracks` endpoint would take on the order of tens of thousands of requests, by contrast, we do about 50 requests now instead with this flow.